knowledge_base/wiki/Code-Style.md

```diff
@@ -0,0 +1,87 @@
+This is the starter page for the Commonwealth Code Style Guide. 
+
+Unless otherwise noted, code styles in this guideline are for NodeJS & TypeScript. It is expected that engineers are already familiar with and follow the official [[TypeScript style guide](https://github.com/microsoft/TypeScript/wiki/Coding-guidelines)](https://github.com/microsoft/TypeScript/wiki/Coding-guidelines) which this document supplements. 
+
+For a list of React-specific guidelines, see [React Best Practices](./React-Best-Practices-And-Improvements.md).
+
+**Contents**
+- [Imports](#imports)
+- [Inline Documentation](#inline-documentation)
+  + [TSDoc](#tsdoc)
+  + [Inline Comments](#inline-comments)
+- [Change Log](#change-log)
+
+# Imports
+
+On server, use relative imports, not absolute: `import foo from '../foobar';`
+
+On client, only some absolute imports are valid. Do not prefix with:
```

Comment:
Should we be changing the codebase to avoid this? Currently it is enabled in the webpack.base.config.js:

```
  resolve: {
    ....
    modules: [
      '../client/scripts',
      '../client/styles',
      '../shared',
      'node_modules', // local node modules
      '../node_modules', // global node modules
    ],
...
```

Reply:
Great question. 

I would presume that we should match our webpack settings against our style guide, using the style guide as our source of truth. 

But these recommendations were originally authored by @CowMuon in January, and being nearly a year old, it would be great to get confirmation that this is still codebase "gospel"

Reply:
Correct. If there is a specific reason why webpack configuration has drifted from code requirements (aka code style guide) we should call it out, or if that was not done for a specific reason, that should be fixed. 

NB. Now that we have adopted RQ for state management which impacts how we do code splitting. In particular RQ will enable us to import more dynamically which means we can have smaller bundles loaded on demand. We could also split our queries & mutations into separate bundles (which seems like something @mzparacha & @kurtisassad should discuss.)

Reply:
Two things to note here:
1. Our style guide drifted with our webpack config, due to the git blame attributing this portion to 2021. I would be fine with changing this to remove the resolve paths, but I don't see a reason to do so currently because it would require a codebase refactor. I don't see the code quality we would gain, seeing as IDEs have auto-importing magic.
2. We still have a ways to go in breaking out the state.ts file. It still points to too many different files leading to the huge circular dependency mess. As soon as we resolve this we can start making informed decisions about where we can code split (see attached image).

![image](https://github.com/hicommonwealth/commonwealth/assets/14794654/dee8235c-cd7b-4d3a-86b7-5eab45427cfa)

Reply:
Should this be split up as a separate ticket, then?

Reply:
Well I think either we:
1. We change the code to match the docs (separate ticket, will resolve convo, will approve this PR)
2. We change the docs to match the code (change needs to be made in this PR, then will approve)

Any preference @ForestMars , since you were the one writing that doc.